backends/nats/nats.go

```diff
@@ -0,0 +1,486 @@
+	// DefaultNatsBucketReplicas - Default replicas for bucket
+	DefaultNatsBucketReplicas = 1
+	// DefaultMaxWaitSeconds - Default max wait for Jetstream replies
+	DefaultMaxWaitSeconds = 10
+)
+
+// internal enum to track auth type
+type authType int
+
+const (
+	username authType = iota
+	token
+	nkey
+	credentials
+	nkeyJwt
+	jwt
+	invalid
+)
+
+// Options enumerates various configuration options
+type Options struct {
+	// Params related to reconnection attempts
+	DisableRetryOnFailedConnect bool `yaml:"disableRetryOnFailedConnect"`
+	MaxReconnects               int  `yaml:"maxReconnects"`
+	ReconnectWaitSeconds        int  `yaml:"reconnectWait"`
+	MaxWaitSeconds              int  `yaml:"maxWaitSeconds"`
+	// Authentication options
+	NatsUsername            string `yaml:"natsUsername"`
+	NatsPassword            string `yaml:"natsPassword"`
+	NatsToken               string `yaml:"natsToken"`
+	NatsNkeySeedFilePath    string `yaml:"natsNkeySeedFilePath"`
+	NatsCredentialsFilePath string `yaml:"natsCredentialsFilePath"`
+	NatsJWTKeyFilePath      string `yaml:"natsJWTKeyFilePath"`
+	// TLS Options
+	// NatsTLSRootCA Specify a custom root CA file (e.g. for self-signed certs)
+	NatsTLSRootCA string `yaml:"natsTLSRootCA"`
+	// NatsTLSClientCert Specify a client cert file
+	NatsTLSClientCert string `yaml:"natsTLSClientCert"`
+	// NatsTLSClientKey Specify a client key file
+	NatsTLSClientKey string `yaml:"natsTLSClientKey"`
+	// Connection options
+	NatsURL string `yaml:"natsURL"`
+	// Storage Options
+	// NatsBucket defines the bucket name
+	NatsBucket string `yaml:"natsBucket"`
+	// NatsBucketReplicas defines number of replicas
+	NatsBucketReplicas int `yaml:"natsBucketReplicas"`
+	// NatsBucketDescription description for NatsBucket
+	NatsBucketDescription string `yaml:"natsBucketDescription"`
+	// CreateBucket tells us to try to create the bucket
+	CreateBucket bool `yaml:"create_bucket"`
+	// CreateBucketPlacementCluster optionally set the cluster placement value when creating a bucket
+	CreateBucketPlacementCluster string `yaml:"createBucketPlacementCluster"`
+	// CreateBucketPlacementTagList comma seperated list of optional placement tags when creating a bucket
+	CreateBucketPlacementTagList string `yaml:"createBucketTagList"`
+	// GlobalPrefix is a prefix applied to all operations, allowing work within a prefix
+	GlobalPrefix string `yaml:"global_prefix"`
+	// EncryptionKey -  key (hex format,256 bit) for encryption at rest
+	EncryptionKey string `yaml:"encryptionKey"`
+	// S3 Compatability
+	// PrefixFolders can be enabled to make List operations show nested prefixes as folders
+	// instead of recursively listing all contents of nested prefixes
+	PrefixFolders bool `yaml:"prefix_folders"`
+	// Internally managed options
+	// Auth type flag set internally by checkCredentialsAvailability()
+	internalUseAuthType authType
+	// Converted type for wait seconds
+	internalReconnectWaitSeconds time.Duration
```

Comment:
Don't add Seconds to the name if it's a Duration.

As mentioned elsewhere, there is no need to convert. You can just directly use `time.Duration.`